### ISSUE TITLE
Support id_token in refresh_token

### ISSUE THREAD

**Issue by making** [cloudfoundry/uaa#777]
### What version of UAA are you running?

``` json
{
  "app": {
    "version": "4.10.0"
  },
  "links": {
    "uaa": "https://uaa.<...>",
    "login": "https://login.<...>"
  },
  "zone_name": "uaa",
  "entityID": "login.<...>",
  "commit_id": "a4132cb",
  "idpDefinitions": {},
  "prompts": {
    "username": [
      "text",
      "Email"
    ],
    "password": [
      "password",
      "Password"
    ]
  },
  "timestamp": "2018-02-06T10:24:36-0800"
}
```

### How are you deploying the UAA?

using cf-deployment 1.15.0 on bosh-lite

### What did you do?

I'm trying to integrate UAA and Kubernetes with CFCR 0.14.0 using https://github.com/cloudfoundry-incubator/kubo-release/pull/101 .

It worked to access k8s with `id_token` returned by UAA.
After the token expired, `kubctl` used `refresh_token` to refresh the token but failed with the following error

```
Unable to connect to the server: token response did not contain an id_token, either the scope "openid" wasn't requested upon login, or the provider doesn't support id_tokens as part of the refresh response.
```

From [the doc](https://docs.cloudfoundry.org/api/uaa/version/4.8.0/index.html#refresh-token),
refresh token does not seem to support id token.

I've also confirmed the response of bellow didn't contain `id_token` field

```
curl -k -v https://uaa.<...>/oauth/token -d client_id=xxxxx -d client_secret=xxxxx -d grant_type=refresh_token -d refresh_token=xxxxx -d response_type=id_token
``` 

### What did you expect to see? What goal are you trying to achieve with the UAA?

I expected UAA to support id_token in refresh_token in order to refresh token with `kubectl`.

**Comment by making** [comment]
@tnwang 

Can you please prioritise this? I believe this is important for k8s integration.

https://kubernetes.io/docs/admin/authentication/#option-1-oidc-authenticator

> Once your id_token expires, kubectl will attempt to refresh your id_token using your refresh_token and client_secret storing the new values for the refresh_token and id_token in your .kube/config.

**Comment by tnwang** [comment]
Thanks for raising this issue. We are aware of this and will be working on it soon.

**Comment by making** [comment]
is this issue fixed in 4.20.0?

**Comment by jhamon** [comment]
@making No, it's not fixed in 4.20.0. I have work in progress for this that I plan to PR for the OD-PKS team.

**Comment by shamus** [comment]
Closing due to inactivity.
